### PR TITLE
[refine](column)  ColumnArray does not implement the insert data function.

### PR DESCRIPTION
**Pull request by Mryange**
### What problem does this PR solve?

ColumnArray cannot determine how many elements it contains from a single block of memory. 
The original approach where the result of get data at cannot be inserted back 
into insert data. Therefore, this function is not implemented directly.
### Release note

None

### Check List (For Author)

- Test 
    - [ ] Regression test
    - [ ] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [x] No need to test or manual test. Explain why:
        - [x] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason 

- Behavior changed:
    - [x] No.
    - [ ] Yes. 

- Does this need documentation?
    - [x] No.
    - [ ] Yes. 

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

File: be/src/vec/columns/column_dictionary.h [hunk 1]
```diff
@@ -158,10 +158,6 @@ class ColumnDictionary final : public COWHelper<IColumn, ColumnDictionary<T>> {
         __builtin_unreachable();
     }
 
-    bool is_fixed_and_contiguous() const override { return true; }
```
Comment (amorynan) [inline be/src/vec/columns/column_dictionary.h:161]: is this can be deleted ? 
because get_raw_data() maybe depends on it?!

Comment (Mryange) [reply be/src/vec/columns/column_dictionary.h:161]: now , only use in array.

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Review by zhangstar333 — APPROVED** [review]

**Review by Gabriel39 — APPROVED** [review]

**Pull request merged**